File: Python/846_Steps.py
from sys import stdin, setrecursionlimit
setrecursionlimit(1000000)

memo = dict()

def solve(x, y, step):
    global memo
    ans = None
    if (x,y,step) in memo:
        ans = memo[(x,y,step)]
    else:
        if x >= y:
            ans = 0
        elif x+step >= y:
            ans = 1
        else:
            ans = 2 + solve(x+step, y-step, step+1)
        memo[(x,y,step)] = ans
    return ans

def main():
    N = int(stdin.readline())
    for n in range(N):
        x, y = map(int, stdin.readline().split())
        print(solve(x, y, 1))
main()

"""
1
45 50
"""

File: UVa Online Judge/Cpp/846_Steps.cpp
#include <iostream>

int solve(int x, int y, int step)
{
    int ans = 0;
    if (x >= y)
    {
        ans = 0;
    }else if (x+step >= y){
        ans = 1;
    }else{
        ans = 2+solve(x+step, y-step, step+1);
    }
    return ans;
}


int main()
{
    int N;
    std::cin>>N;
    for (int n = 0; n < N ; n++)
    {
        int x, y;
        std::cin>>x>>y;
        std::cout<<solve(x,y,1)<<std::endl;
    }
    return 0;
}